src/docfx/build/xref/XrefMap.cs

```diff
@@ -154,22 +151,45 @@ private static XrefSpec LoadSchemaDocument(List<Error> errors, JToken token, Doc
                     throw Errors.SchemaNotFound(file.Mime).ToException();
                 }
 
+                var uidExists = schema.Type.GetProperty("Uid") != null;
                 var (schemaErrors, content) = JsonUtility.ToObject(token, schema.Type, transform: TransformContent);
                 errors.AddRange(schemaErrors);
-                return xref;
+                if (!string.IsNullOrEmpty((string)schema.Type.GetProperty("Uid")?.GetValue(content, null)))
+                {
+                    return BuildXref();
+                }
+                else if (extensionData.Count > 0)
+                {
+                    errors.Add(Errors.UidMissing());
+                }
+                return null;
 
-                object TransformContent(DataTypeAttribute attribute, object value)
+                XrefSpec BuildXref()
```

Comment:
populate a `JObject` from xref properties, then deserialize it into an `XrefSpec`, to avoid manually parse `uid`, `name`, `href` etc